x/auction/types/auctions.go

```diff
@@ -66,15 +68,32 @@ func (a BaseAuction) GetType() string { return "base" }
 
 // Validate verifies that the auction end time is before max end time
 func (a BaseAuction) Validate() error {
-	if a.EndTime.After(a.MaxEndTime) {
-		return fmt.Errorf("MaxEndTime < EndTime (%s < %s)", a.MaxEndTime, a.EndTime)
+	// ID can be 0 for surplus, debt and collateral auctions
+	if strings.TrimSpace(a.Initiator) == "" {
+		return errors.New("auction initiator cannot be blank")
 	}
 	if !a.Lot.IsValid() {
 		return fmt.Errorf("invalid lot: %s", a.Lot)
 	}
+	// NOTE: bidder can be empty for Surplus and Collateral auctions
+	if !a.Bidder.Empty() && len(a.Bidder) != sdk.AddrLen {
+		return fmt.Errorf("the expected bidder address length is %d, actual length is %d", sdk.AddrLen, len(a.Bidder))
+	}
 	if !a.Bid.IsValid() {
 		return fmt.Errorf("invalid bid: %s", a.Bid)
 	}
+	if a.EndTime.IsZero() || a.MaxEndTime.IsZero() {
+		return errors.New("end time cannot be zero")
+	}
+	if a.EndTime.After(a.MaxEndTime) {
+		return fmt.Errorf("MaxEndTime < EndTime (%s < %s)", a.MaxEndTime, a.EndTime)
+	}
+	if a.HasReceivedBids && !a.Bid.IsPositive() {
+		return fmt.Errorf("cannot have a zero value bid when auction has a true HasReceivedBids flag value")
+	}
+	if !a.HasReceivedBids && a.Bid.IsPositive() {
+		return fmt.Errorf("cannot have a positive value bid when auction has a false HasReceivedBids flag value")
+	}
```

Comment:
These checks aren't valid for all the auction types, I would probably remove them (though you could do different checks for each auction type if you realy wanted).

Bad naming on my part, but "bidding" on an auction doesn't necessarily mean the "Bid" value is increasing:

- SurplusAuction:    Lot stays constant, Bid increases on each MsgPlaceBid
- DebtAuction:        Bid stays constant, Lot decreases on each MsgPlaceBid
- CollateralAuction: Lot stays constant, while Bid increases up to a max, at which point Bid stays constant and Lot decreases